src/libstd/io/buffered.rs

```diff
@@ -752,11 +752,13 @@ impl<W> fmt::Display for IntoInnerError<W> {
 ///     let file = File::create("poem.txt")?;
 ///     let mut file = LineWriter::new(file);
 ///
-///     for &byte in road_not_taken.iter() {
-///        file.write(&[byte]).unwrap();
-///     }
+///     file.write_all(road_not_taken)?;
```

Comment:
The example is explicitly writing each byte individually to note how the buffering provided by LineWriter helps things. If we were going to write the whole message out at once, there's no reason to use the LineWriter at all.

Reply:
It's still line buffering behind the scenes, so I don't think that it's totally wrong. I'm certainly open to suggestions on how to improve the entire example, though. Maybe we could write some bytes, and demonstrate that they aren't written to disk until a newline is written?

Reply:
That seems reasonable.